wormhole-connect/src/utils/routes/cctpManual.ts

```diff
@@ -0,0 +1,586 @@
+    throw new Error('not implemented');
+  }
+
+  async estimateSendGas(
+    token: TokenId | 'native',
+    amount: string,
+    sendingChain: ChainName | ChainId,
+    senderAddress: string,
+    recipientChain: ChainName | ChainId,
+    recipientAddress: string,
+    routeOptions: any,
+  ): Promise<string> {
+    const provider = wh.mustGetProvider(sendingChain);
+    const { gasPrice } = await provider.getFeeData();
+    if (!gasPrice)
+      throw new Error('gas price not available, cannot estimate fees');
+
+    // only works on EVM
+    const chainContext = wh.getContext(
+      sendingChain,
+    ) as EthContext<WormholeContext>;
+    const tokenMessenger =
+      wh.mustGetContracts(sendingChain).cctpContracts?.cctpTokenMessenger;
+    const circleSender = TokenMessenger__factory.connect(
+      tokenMessenger!,
+      wh.getSigner(sendingChain)!,
+    );
+    const tokenAddr = await wh.mustGetForeignAsset(
+      token as TokenId,
+      sendingChain,
+    );
+    const toChainName = wh.toChainName(recipientChain)!;
+    const decimals = getTokenDecimals(wh.toChainId(sendingChain), token);
+    const parsedAmt = utils.parseUnits(`${amount}`, decimals);
+    const destinationDomain = wh.conf.chains[toChainName]?.cctpDomain;
+    if (destinationDomain === undefined)
+      throw new Error(`CCTP not supported on ${toChainName}`);
+    try {
+      const tx = await circleSender.populateTransaction.depositForBurn(
+        parsedAmt,
+        destinationDomain,
+        chainContext.context.formatAddress(recipientAddress, recipientChain),
+        chainContext.context.parseAddress(tokenAddr, sendingChain),
+      );
+      const est = await provider.estimateGas(tx);
+      const gasFee = est.mul(gasPrice);
+      return toFixedDecimals(utils.formatEther(gasFee), 6);
+    } catch (Error) {
+      return getGasFeeFallback(token, sendingChain, Route.CCTPManual);
+    }
+
+    // maybe put this in a try catch and add fallback!
+  }
+
+  async estimateClaimGas(destChain: ChainName | ChainId): Promise<string> {
+    // Note: This hardcodes 300000 gas - just as is done for the token bridge claim route
+    const provider = wh.mustGetProvider(destChain);
+    const gasPrice = await provider.getGasPrice();
+
+    const est = BigNumber.from('300000');
+    const gasFee = est.mul(gasPrice);
+    return toFixedDecimals(utils.formatEther(gasFee), 6);
+  }
+
+  /**
+   * These operations have to be implemented in subclasses.
+   */
+  async send(
+    token: TokenId | 'native',
+    amount: string,
+    sendingChain: ChainName | ChainId,
+    senderAddress: string,
+    recipientChain: ChainName | ChainId,
+    recipientAddress: string,
+    routeOptions: any,
+  ): Promise<string> {
+    const fromChainId = wh.toChainId(sendingChain);
+    const fromChainName = wh.toChainName(sendingChain);
+    const decimals = getTokenDecimals(fromChainId, token);
+    const parsedAmt = utils.parseUnits(amount, decimals);
+
+    // only works on EVM
```

Comment:
Same as above, throw if not an evm. That way we can quickly detect that something is wrong instead of letting it fail at the subsequent calls